commands/up.go

```diff
@@ -67,17 +77,88 @@ func upHandler(cmd *cobra.Command, args []string) error {
 	if err := runBuild(cmd, args); err != nil {
 		return err
 	}
-	fmt.Println()
+
 	if !skipPush {
 		if err := runPush(cmd, args); err != nil {
 			return err
 		}
-		fmt.Println()
 	}
+
 	if !skipDeploy {
 		if err := runDeploy(cmd, args); err != nil {
 			return err
 		}
 	}
+
+	if watch {
+		logger := log.New(os.Stderr, "", log.LstdFlags)
+		logger.SetFlags(log.Ldate | log.Ltime | log.LUTC)
+
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			return err
+		}
+		defer watcher.Close()
+
+		err = filepath.Walk("./", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return watcher.Add(path)
+			}
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		signalChannel := make(chan os.Signal, 1)
+		signal.Notify(signalChannel, os.Interrupt, syscall.SIGTERM)
+
+		go func() {
+			<-signalChannel
+
+			watcher.Close()
+
+			os.Exit(0)
+		}()
+
+		d := debounce.New(5 * time.Second)
+
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return fmt.Errorf("watcher's Events channel is closed")
+				}
+
+				if event.Op == fsnotify.Write {
+					d(func() {
+						if err := runBuild(cmd, args); err != nil {
+							logger.Fatalf("%s %s", functionName, err)
+						}
+
+						if !skipPush {
+							if err := runPush(cmd, args); err != nil {
+								logger.Fatalf("%s %s", functionName, err)
+							}
+						}
+
+						if err := runDeploy(cmd, args); err != nil {
+							logger.Fatalf("%s %s", functionName, err)
+						}
+					})
```

Comment:
Instead of duplicating this flow, i would consider defining a new function 

```go
// upRunner is constructs a handler function that runs the build, push and deploy commands one time.
func upRunner(cmd *cobra.Command, args []string) func() error {
	return func() error {
		if err := runBuild(cmd, args); err != nil {
			return err
		}

		if !skipPush {
			if err := runPush(cmd, args); err != nil {
				return err
			}
		}

		if !skipDeploy {
			if err := runDeploy(cmd, args); err != nil {
				return err
			}
		}
		return nil
	}
}
```

This is responsible for defining the actual up logic and can be called multiple times, specifically, inside the `upHandler` you can now do this 

```go
	handler := upRunner(cmd, args)

	err := handler()
	if err != nil {
		return err
	}
```

and 

```go
				if event.Op == fsnotify.Write {
					d(func() {
						if err := handler(); err != nil {
							logger.Printf("%s %s", functionName, err)
						}
					})
				}
```

I would avoid the `FatalF` here so that the watch can survive a typo in the code, etc. Let the watch run until the user explicitly stops it. 

You could also consider adding this helper function 

```go
func logError(f func() error) func() {
	return func() {
		if err := f(); err != nil {
			log.Println(err)
		}
	}
}
```

which then allows you to further simplify the debounce to look like this 

```go
				if event.Op == fsnotify.Write {
					d(logError(handler))
				}
```